blocks/site-details/renderSiteSEO.js

```diff
@@ -94,66 +188,97 @@ export default async function renderSiteSEO({ container, nav, renderOptions }) {
-                // eg exclude fragments
-                if (content.toLowerCase().includes('noindex')) {
-                  tableRow.remove();
-                }
-              } else {
-                el.innerHTML = content;
-              }
-            } else {
-              el.innerHTML = '<div class="badge orange">Missing</div>';
-            }
+  // MARK: create rows
+  const createTableRows = (data, startIndex, endIndex) => {
+    let dataToParse;
+    if (typeof startIndex === 'number' && typeof endIndex === 'number') {
+      dataToParse = data.slice(startIndex, endIndex);
+    } else {
+      dataToParse = data;
+    }
+    const rows = dataToParse.reduce((output, item) => {
+      const previewRow = document.createElement('tr');
+      previewRow.dataset.path = item.path;
+      previewRow.dataset.fetchUrl = `${projectDetails.customPreviewUrl}${item.path}`;
+      previewRow.dataset.environment = 'previewed';
+      const publishRow = document.createElement('tr');
+      publishRow.dataset.path = item.path;
+      publishRow.dataset.fetchUrl = `${projectDetails.customLiveUrl}${item.path}`;
+      publishRow.dataset.environment = 'published';
+
+      const initRow = (row) => {
+        // MARK: populate row
+        // assign a populate function to each row
+        row.populate = async () => {
+          // Prevent populating rows that are not displayed
+          if (row.dataset.environment !== container.dataset.showEnvironment) return;
+          // Rows are only removed if you change pages, so check if populated.
+          if (row.dataset.isPopulated === 'true') return;
+          row.dataset.isPopulated = 'true';
+
+          row.innerHTML = `
+            <td data-meta-property="og:image"><div class="skeleton" style="width: 64px; height: 64px;"></div></td>
+            <td class="path"><strong>${safeText(item.path)}</strong></td>  
+            <td data-meta-property="og:title"><div class="skeleton" style="width: 100px; height: 30px;"></div></td>
+            <td data-meta-property="og:description"><div class="skeleton" style="width: 200px; height: 48px;"></div></td>
+            <td data-meta-property="keywords"><div class="skeleton" style="width: 150px; height: 30px;"></div></td>
+            <td class="buttons">
+              <div class="button-container">
+                <a class="button action secondary edit" target="_blank" >Edit</a>
+              </div>
+            </td>
+          `;
+
+          const editButton = row.querySelector('.edit');
+          if (projectDetails.darkAlleyProject) {
+            editButton.href = `https://da.live/edit#/${daProjectRepo}/${siteSlug}/metadata${item.path.endsWith('/') ? `${item.path}/index` : item.path}`;
+          } else {
+            editButton.href = `https://docs.google.com/document/d/${item.id}/edit`;
+          }
+          // Cached fetch in case user comes back to this page through pagination.
+          const response = await cacheFetch(row.dataset.fetchUrl);
```

Comment:
I think the cache from `cacheFetch` should be cleared when a user switches tabs else there is a potential disconnect between the `/index` response and the cached SEO data from the pages WDYT ?

Reply:
Also in the tour you tell the user: 
```
This is a direct link to edit the page. You can adjust your metadata, preview or publish your page and come back to this page.
```
So you could add a mechanism to refresh the displayed data when the tab gains focus again eg with https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilitychange_event but that might be for another issue ...